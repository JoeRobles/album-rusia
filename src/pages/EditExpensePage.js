import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../components/ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.startRemoveExpense(this.props.expense.id);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
          action="Edit"
        />
        <button onClick={this.onRemove}>Remove
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(
    (expense) => expense.id === props.match.params.id,
  ),
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (expenseId) => dispatch(startRemoveExpense(expenseId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
