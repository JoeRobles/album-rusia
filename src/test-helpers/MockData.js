import moment from 'moment';

export const expensesReducerDefaultState = [
  {
    id: 'poijasdfhwer',
    description: 'January Rent',
    note: 'This was the final payment for that address',
    amount: 54500,
    createdAt: 1514954400000,
  },
  {
    id: 'lmdfmlgkdnfg',
    description: 'water bill',
    amount: 100,
    createdAt: 1515085200000,
  },
  {
    id: 'fhgerthhrdhd',
    description: 'gas bill',
    amount: 300,
    createdAt: 1515140400000,
  },
  {
    id: 's√±ojglsjslkg',
    description: 'internet bill',
    amount: 200,
    createdAt: 1515267500000,
  },
];

export const filtersReducerDefaultState = {
  text: '',
  sortBy: 'date',
  asc: true,
  startDate: moment().startOf('month'),
  endDate: moment().endOf('month'),
};