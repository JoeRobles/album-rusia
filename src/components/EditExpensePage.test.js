import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from './EditExpensePage';
import { expensesReducerDefaultState } from '../test-helpers/MockData';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditExpensePage
      editExpense={editExpense}
      removeExpense={removeExpense}
      history={history}
      expense={expensesReducerDefaultState[0]}
    />
  );
});

it('should render editExpensePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle editExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expensesReducerDefaultState[0]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(editExpense).toHaveBeenLastCalledWith(expensesReducerDefaultState[0].id, expensesReducerDefaultState[0]);
});

it('should handle removeExpense', () => {
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(removeExpense).toHaveBeenLastCalledWith(expensesReducerDefaultState[0].id);
});
