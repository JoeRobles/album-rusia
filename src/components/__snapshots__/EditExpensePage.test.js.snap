// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 54500,
        "createdAt": 1514954400000,
        "description": "January Rent",
        "id": "poijasdfhwer",
        "note": "This was the final payment for that address",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
